### PR TITLE
feat(hub-common): add site URL to settings schema

### PR DESCRIPTION
**Pull request by tomwayson**
affects: @esri/hub-common

1. Description:

1. Instructions for testing:

1. Closes Issues: #<number> (if appropriate)

1. [x] Updated meaningful TSDoc to methods including Parameters and Returns, see [Documentation Guide](https://esri.github.io/hub-components/storybook/?path=/story/guides-documentation--page)

1. [x] used semantic commit messages
  
1. [x] PR title follows semantic commit format (**CRITICAL** if the title is not in a semantic format, the release automation will not run!)

1. [x] updated `peerDependencies` as needed. **CRITICAL** our automated release system can **not** be counted on to update `peerDependencies` so we _must_ do it manually in our PRs when needed. See the [updating peerDependencies](/RELEASE.md#Updating-peerDependencies) section of the release instructions for more details.

### CODE REVIEW

File: packages/common/src/core/types/IHubSite.ts [hunk 1]
```diff
@@ -7,12 +7,30 @@ import { IWithCatalog, IWithCatalogs } from "../traits/IWithCatalog";
 import { IWithLayout } from "../traits/IWithLayout";
 import { IWithPermissions } from "../traits/IWithPermissions";
 
+export interface IHubSiteUrlInfo {
```
Comment (dbouwman) [inline packages/common/src/core/types/IHubSite.ts:10]: maybe add a comment that while not used for any other entity (thus why it's in this file) this is extracted into an interface to ensure IHubSiteEditor shares a consistent structure (this took a me little 🤔)

File: packages/common/src/sites/_internal/SiteSchema.ts [hunk 2]
```diff
@@ -17,11 +17,17 @@ export const SiteEditorTypes = [
...
+  ({
+    $async: true,
+    ...HubItemEntitySchema,
+    properties: {
+      ...HubItemEntitySchema.properties,
+      _discussions: ENTITY_IS_DISCUSSABLE_SCHEMA,
+      telemetry: PRIVACY_CONFIG_SCHEMA,
+      _urlInfo: {
+        type: "object",
+        isUniqueDomain: { siteId },
```
Comment (dbouwman) [inline packages/common/src/sites/_internal/SiteSchema.ts:30]: I'm far from a json schema expert, but how does this work when creating a new site, when there is no siteId?

File: packages/common/test/sites/_internal/SiteuUiSchemaTelemetry.test.ts [hunk 3]
```diff
@@ -1,15 +1,44 @@
 import { buildUiSchema } from "../../../src/sites/_internal/SiteUiSchemaSettings";
 import { MOCK_CONTEXT } from "../../mocks/mock-auth";
 
-describe("buildUiSchema: site telemetry", () => {
-  it("returns the full site telemetry uiSchema", async () => {
-    const uiSchema = await buildUiSchema("some.scope", {} as any, MOCK_CONTEXT);
+describe("buildUiSchema: site settings", () => {
```
Comment (dbouwman) [inline packages/common/test/sites/_internal/SiteuUiSchemaTelemetry.test.ts:4]: Since you renamed the test, should the file name also change?

**Review by dbouwman — APPROVED** [review]
Code looks good - but make sure this works w/ the "create site" flow b/c I'm not clear how getSiteSchema works if there is no siteId...

File: packages/common/src/sites/_internal/SiteSchema.ts (hunk 2 above)
Comment (tomwayson) [reply packages/common/src/sites/_internal/SiteSchema.ts:30]: that's a good question.

short answer: it's fine

long answer: the keyword validator that this schema is passed into only uses the `siteId` if it was passed in. Here's the logic:

```ts
  // siteId is only passed in when editing an existing site
  const siteId = schema?.siteId;
  let isValid;
  try {
    const domain = await lookupDomain(hostname, context.hubRequestOptions);
    // if lookupDomain does NOT throw, we found a domain record
    // which is only valid if it is the one associated with the site we are editing
    isValid = !!siteId
      ? domain.siteId === siteId
      : false;
  } catch (_) {
    // no domain found, so it's available
    isValid = true;
  }
  return isValid;
```

File: packages/common/test/sites/_internal/SiteuUiSchemaSettings.test.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
+import { buildUiSchema } from "../../../src/sites/_internal/SiteUiSchemaSettings";
```
Comment (juliannemarik) [inline packages/common/test/sites/_internal/SiteuUiSchemaSettings.test.ts:1]: minor: looks like there's still a small typo in the file name? Should "SiteuUiSchemaSettings.test.ts" should be "SiteUiSchemaSettings.test.ts"?

**Review by juliannemarik — APPROVED** [review]
This all looks good to me!

**Pull request merged**